node/runtime/src/lib.rs

```diff
@@ -436,11 +439,11 @@ pub type SignedBlock = generic::SignedBlock<Block>;
 /// BlockId type as expected by this runtime.
 pub type BlockId = generic::BlockId<Block>;
 /// The SignedExtension to the basic transaction logic.
-pub type SignedExtra = (system::CheckNonce<Runtime>, balances::TakeFees<Runtime>);
+pub type SignedExtra<R> = (system::CheckNonce<R>, system::CheckWeight<R>, balances::TakeFees<R>);
```

Comment:
Why?

Reply:
Because in different places (`executive`, `executor` etc.), we also want to know the _real runtime's transaction extension format_ and based on the environment, if, in `TestExternalities`, we can't import the `SignedExtra` without the generic, and I don't want to duplicate the code there.

Reply:
https://github.com/paritytech/substrate/pull/3115/files#diff-9be02fd5e4de6c77f22d6327fdaa09afL438

Reply:
that sentence isn't english! please reformulate.

Reply:
if what you're getting is what i think it is, then it's an incorrect assertion anyway. the node runtime's `SignedExtra` type is only meaningful when it's bound to the node runtime's `Runtime` type. the example you gave binds it to some other `Runtime` type, taking it out of context. in that case, the text runtime should have its own `SignedExtra` (which may look fairly similar, just as many of the config traits do, but isn't semantically deduplicatable).

Reply:
> in that case, the test runtime should have its own SignedExtra (which may look fairly similar, just as many of the config traits do, but isn't semantically deduplicatable).

Ok this makes sense. I prioritized not having duplicated code as much as possible but I see that it was not correct. The rest of your comments all point back to this matter, which I will revert. 👍